api/pkg/transformer/server/grpc/server.go

```diff
@@ -53,6 +57,7 @@ type UPIServer struct {
 	opts                  *config.Options
 	modelClient           upiv1.UniversalPredictionServiceClient
 	conn                  grpcpool.ConnPool
+	httpClient            hystrixHttpClient
```

Comment:
Instead of having both httpClient and modelClient, would it be possible to replace both of them with a single interface? We can instantiate different implementation of the interface depending on whether UPI on http is enabled or not.

Reply:
that's a good idea. I already updated this